src/modules/didRegistry/didRegistry.validation.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  DIDAttribute,
+  Encoding,
+  PubKeyType,
+} from '@ew-did-registry/did-resolver-interface';
+import { KeyType } from '@ew-did-registry/keys';
+import { createCheckers, iface, enumtype } from 'ts-interface-checker';
+
+const IS_REQ_STRING = 'string';
+export const { UpdateServicePoint, UpdateDelegate, UpdatePublicKey } =
```

Comment:
Autogenerated checker would only verify if object is aligned with it's TS definition. It won't validate anything more than type of props. 
As far as I'm concerned, what we need is:
- conditional validation depending on `DIDAttribute` - Claim, PublicKey or Delegate
- confirmation if specific enum is of specific value
- depending on didAttribute some of optional props are in fact required
The only way we could achieve it with autogeneration is to 
- split Claim, PublicKey and Delegate into 3 separate interfaces, but this would require from us a lot of effort and probably would be against the initial architectural decisions. 
- or we could generate 3 interfaces only for the purpose of validation, but this would be more confusing than current solution.

Reply:
Make sense